workstation-backend/tests/component/test_views.py

```diff
@@ -0,0 +1,236 @@
+import json
+import pytest
+
+from decimal import Decimal
+from uuid import uuid4
+
+from django.urls import reverse
+from rest_framework import status
+
+from component.models import PostIt
+
+
+class BaseTestPostIt:
+    def _assert_result(self, result, expected_post_it):
+        assert result["id"] == str(expected_post_it.id)
+        assert result["workspace_id"] == str(expected_post_it.workspace.id)
+        assert Decimal(result["x"]) == Decimal(expected_post_it.x)
+        assert Decimal(result["y"]) == Decimal(expected_post_it.y)
+        assert Decimal(result["height"]) == Decimal(expected_post_it.height)
+        assert Decimal(result["width"]) == Decimal(expected_post_it.width)
+        assert result["collapsed"] == expected_post_it.collapsed
+        assert result["title"] == expected_post_it.title
+        assert result["content"] == expected_post_it.content
+        assert result["color"] == expected_post_it.color
+
+
+@pytest.mark.django_db
+class TestPostitList(BaseTestPostIt):
+    def test_get_not_found(self, client):
+        url = reverse("post-it-list", kwargs={"workspace_id": uuid4()})
+
+        response = client.get(url)
+        assert response.status_code == status.HTTP_200_OK
+        assert response.json() == []
+
+    def test_get(self, client, post_it_1, post_it_2):
+        url = reverse("post-it-list", kwargs={"workspace_id": post_it_1.workspace.id})
+
+        response = client.get(url)
+        assert response.status_code == status.HTTP_200_OK
+
+        result = response.json()
+        assert len(result) == 1
+        self._assert_result(result[0], post_it_1)
+
+        url = reverse("post-it-list", kwargs={"workspace_id": post_it_2.workspace.id})
+
+        response = client.get(url)
+        assert response.status_code == status.HTTP_200_OK
+
+        result = response.json()
+        assert len(result) == 1
+        self._assert_result(result[0], post_it_2)
+
+    def test_post(self, client, workspace_1):
+        url = reverse("post-it-list", kwargs={"workspace_id": workspace_1.id})
+
+        response = client.post(
+            url,
+            data={
+                "title": "New Post",
+                "content": "Sample content",
+                "x": 0,
+                "y": -100,
+                "width": 20.25,
+                "height": 52.02,
```

Comment:
Looking at this, I realized the following:

- Are we allowing negative coordinate?
- Are we allowing coordinates or dimensions larger than the window size?

Reply:
I think we should keep this flexible (i.e. allow negative coordinate + any value) for now since we haven't set up the front end. Once the front end implementation is in progress we can revisit this to add more constraint.